packages/core/src/service/services/order.service.ts

```diff
@@ -441,6 +448,27 @@ export class OrderService {
         this.eventBus.publish(new OrderStateTransitionEvent(fromState, state, ctx, order));
         return order;
     }
+    async findOrderByFulfillmentId(fulfillment: Fulfillment): Promise<Order> {
```

Comment:
I think it might make sense to extract all these fulfillment-related methods into their own service now. The OrderService is already pretty huge, and now there's enough specific Fulfillment stuff to justify it.